types/underscore/underscore-tests.ts

```diff
@@ -165,372 +165,423 @@ declare const level2UnionList: _.List<_.List<string | number>>;
-var renderNotes = _.after(notes.length, render);
-_.each(notes, (note) => note.asyncSave({ success: renderNotes }));
+{
+    const renderNotes = _.after(anyArray.length, () => alert("rendering...")); // $ExpectType Function
+    _.each(anyArray, note => note.asyncSave({ success: renderNotes })); // $ExpectType any[]
+}
 
 // wrapping a function in another function
-var hello = function (name) { return "hello: " + name; };
-var wrappedHello = _.wrap(hello, (func) => { return `before, ${func("moe")} + after`; });
-wrappedHello();
+{
+    const hello = (name: string) => "hello: " + name;
+    const hello2 = _.wrap(hello, func => `before, ${func("moe")} + after`); // $ExpectType Function
+    hello2(); // $ExpectType any
+}
 
 // composing a function as the result of multiple function calls
-var greet = function (name) { return "hi: " + name; };
-var exclaim = function (statement) { return statement + "!"; };
-var welcome = _.compose(exclaim, greet);
-welcome('moe');
-
-var partialApplicationTestFunction = (a: string, b: number, c: boolean, d: string, e: number, f: string) => {  }
+{
+    const greet = (name: string) => "hi: " + name;
+    const exclaim = (statement: string) => statement + "!";
+    const welcome = _.compose(exclaim, greet); // $ExpectType Function
+    welcome('moe'); // $ExpectType any
+}
 
 // providing a partial set of leading arguments
-var partialApplicationResult = _.partial(partialApplicationTestFunction, "", 1);
+_.partial(manyParameters, "", 1); // $ExpectType (p3: boolean, p4: string, p5: number, p6: string) => string
 
 // providing a partial set of arguments in the middle of a parameter set
-var parametersCanBeStubbed = _.partial(partialApplicationResult, _, _, _, "");
+_.partial(manyParameters, _, _, _, ""); // $ExpectType (p1: string, p2: number, p3: boolean, p5: number, p6: string) => string
 
 ///////////////////////////////////////////////////////////////////////////////////////
 
 // retrieving the keys of an object
-_.keys({ one: 1, two: 2, three: 3 });
+_.keys(explicitNumberDictionary); // $ExpectType string[]
 
 // retrieving the values of a dictionary
-_.values({ one: 1, two: 2, three: 3 });
+_.values(explicitNumberDictionary); // $ExpectType number[]
 
 // retrieving an array of key-value pairs for an object
-_.pairs({ one: 1, two: 2, three: 3 });
+_.pairs(explicitNumberDictionary); // $ExpectType ["one" | "two" | "three", number][]
 
 // making an object's keys its values and values its keys
-_.invert({ Moe: "Moses", Larry: "Louis", Curly: "Jerome" });
+_.invert({ Moe: "Moses", Larry: "Louis", Curly: "Jerome" }); // $ExpectType any
 
 // retrieving the names of all of the function-valued properties from an object
-_.functions(_);
+_.functions(_); // $ExpectType string[]
 
 // shallow copying properties from the source objects to the destination object
-_.extend({ name: 'moe' }, { age: 50 });
+_.extend({ name: 'moe' }, { age: 50 }, { userid: 'moe1' }); // $ExpectType any
```

Comment:
Hmm, actually not sure now why I added this extra parameter to a few of these, glad to hear it's better to have done so though 😄

Reply:
I think it's better this way, because people (including myself) often overlook the fact that you can use `_.extend` et al with more than one source object.